db/memtable.cc

```diff
@@ -778,11 +779,16 @@ static bool SaveValue(void* arg, const char* entry) {
         return true;
       }
       default: {
-        std::string msg("Unrecognized value type: " +
-                        std::to_string(static_cast<int>(type)) + ". ");
-        msg.append("User key: " + user_key_slice.ToString(/*hex=*/true) + ". ");
-        msg.append("seq: " + std::to_string(seq) + ".");
-        *(s->status) = Status::Corruption(msg.c_str());
+        if (s->log_corruption_errors) {
+          std::string msg("Corrupted value not expected.");
```

Comment:
Same, I think this part of the message is fine to log regardless of the option's value.